### PR TITLE
Instantiate model only once in pipeline

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

The current implementation of `pipeline` is inefficient in the sense it instantiates the model twice just to guess the proper framework. This PR does not add any breaking change but reworks the function that infers the framework from the model to:
1. instantiate the proper class of the model (this avoids getting weird warnings about missing weights)
2. return the model instantiated so it's not re-instantiated later on.

cc @mfuntowicz  and @Narsil

### CODE REVIEW

File: src/transformers/pipelines/base.py
Comment (LysandreJik) [inline src/transformers/pipelines/base.py:54]:
```suggestion
    If :obj:`model` is instantiated, this function will just infer the framework from the model class. Otherwise
```

**Review by LysandreJik — APPROVED** [review]
Yes, LGTM! Let's wait for @Narsil to take a look at this as it will have a direct impact on the inference API.

**Review by Narsil — APPROVED** [review]
LGTM.

We probably should add a test for the user warning, and checking that model loading only happens once.
Also can we run all the slow pipelines test for this one, to be sure ?

**Comment by sgugger** [comment]
Checked all the slow tests run. I have no idea on how to implement a test that checks the model is only loaded once, so I'm going merge this and if anyone wants to tackle that, it can be done in a separate PR.

**Pull request merged**
